connector-packager/connector_packager/jar_jdk_packager.py

```diff
@@ -55,7 +55,13 @@ def get_min_support_version(file_list: List[ConnectorFile], cur_min_version_tabl
                 min_version_tableau = "2020.3"
             reasons.append("Connector uses Connection Dialogs V2, which was added in the 2020.3 release")
         elif connector_file.file_type == "oauth-config":
-            if 2021.1 > float(min_version_tableau):
+            # Check to see if we're using oauthConfigId, which needs 2023.1+
+            pluginOAuthConfigRoot = ET.parse(input_dir / connector_file.file_name).getroot()
+            oauthConfigId = pluginOAuthConfigRoot.find('.//oauthConfigId')
+            if (oauthConfigId is not None):
+                min_version_tableau = "2023.1"
```

Comment:
`oauthConfigId` has existed in some form since, XSD at least, since 21.4. Normally the SDK takes the most permissive stance for compatibility. I'm unsure from the context I have what the right choice is here. 

Think about an existing connector with a min-tableau-version of 21.4 and beginning to adopt this feature. Is this a breaking change in context of those older products, or just graceful fallback to default/existing behavior.

Reply:
FWIW, I don't see `oauthConfigId` mentioned in the documentation or the samples on the SDK, and up until the SDK's XSD were change in this PR connectors using it would have been rejected, so I think the odds of a partner connector having this element before 2023.1 would be very small. Still worth confirming that we're not breaking existing connectors, though.

Reply:
I was thinking about this within the context of adding functionality to a connector that already exists. I'll defer to the team on what the expected future usage is and if this is right choice.

Reply:
After talking more with my team, we decided to change it to 21.4.  While we are still not sure if it will work at the moment, we decided it's better to keep it loose rather than restricting it to 23.1+.  I believe there will be some follow-up stories to ensure that earlier versions can work.